vnext/ReactWindowsCore/WinRTWebSocketResource.cpp

```diff
@@ -354,6 +390,45 @@ void WinRTWebSocketResource::Synchronize()
   }
 }
 
+string WinRTWebSocketResource::GetRestrictedErrorMessage() noexcept
+{
+  string result;
+  CComPtr<IRestrictedErrorInfo> errorInfo;
+  if (SUCCEEDED(GetRestrictedErrorInfo(&errorInfo)))
+  {
+    BSTR description;
+    HRESULT error;
+    BSTR restrictedDesc;
+    BSTR capabilitySid;
+    if (SUCCEEDED(errorInfo->GetErrorDetails(&description, &error, &restrictedDesc, &capabilitySid)))
+    {
+      try
+      {
+        result = Utf16ToUtf8(description) + ": " + Utf16ToUtf8(restrictedDesc);
+      }
+      catch (const std::exception& e)
+      {
+        result = e.what();
+      }
+    }
+    else
+    {
+      result = "Unknown error";
+    }
+    // BSTR's allocated memory must be explicitly released.
+    // See https://docs.microsoft.com/en-us/cpp/atl-mfc-shared/allocating-and-releasing-memory-for-a-bstr
+    SysFreeString(description);
+    SysFreeString(restrictedDesc);
+    SysFreeString(capabilitySid);
```

Comment:
these get leaked (i.e. these lines aren't run) if something other than std::exception gets thrown

Reply:
We must not use the raw functions like this directly in modern C++.

---
In reply to: [440555346](https://github.com/microsoft/react-native-windows/pull/5174#discussion_r440555346) [](ancestors = 440555346)

Reply:
@vmoroz what is the suggested pattern?
Do we have an available type that can safely wrap `BSTR`?

Reply:
To ensure proper resource handling we should use RAII classes/wrappers.